hnn_core/gui/_viz_manager.py

```diff
@@ -324,33 +330,44 @@ def _plot_on_axes(b, widgets_simulation, widgets_plot_type,
     widgets_plot_type.disabled = True
 
     single_simulation = data['simulations'][sim_name]
-
-    plot_config = {
+    target_is_sim = True if single_simulation['net'] is not None else False
+    scaling = dipole_scaling.value if target_is_sim else data_scaling.value
+    smooth = dipole_smooth.value if target_is_sim else data_smooth.value
+    simulation_plot_config = {
         "max_spectral_frequency": max_spectral_frequency.value,
-        "dipole_scaling": dipole_scaling.value,
-        "dipole_smooth": dipole_smooth.value,
+        "dipole_scaling": scaling,
+        "dipole_smooth": smooth,
         "spectrogram_cm": spectrogram_colormap_selection.value
     }
 
     dpls_processed = _update_ax(fig, ax, single_simulation, sim_name,
-                                plot_type, plot_config)
+                                plot_type, simulation_plot_config)
 
     # If target_simulations is not None and we are plotting a dipole,
     # we need to plot the target dipole as well.
-    if target_simulations.value in data['simulations'].keys(
+    if data_widget.value in data['simulations'].keys(
     ) and plot_type == 'current dipole':
 
-        target_sim_name = target_simulations.value
+        target_sim_name = data_widget.value
         target_sim = data['simulations'][target_sim_name]
+        target_is_sim = True if target_sim['net'] is not None else False
+        scaling = dipole_scaling.value if target_is_sim else data_scaling.value
+        smooth = dipole_smooth.value if target_is_sim else data_smooth.value
+        data_plot_config = {
+            "max_spectral_frequency": max_spectral_frequency.value,
+            "dipole_scaling": scaling,
+            "dipole_smooth": smooth,
+            "spectrogram_cm": spectrogram_colormap_selection.value
+        }
 
         # plot the target dipole.
         # disable scaling for the target dipole.
-        plot_config['dipole_scaling'] = 1.
+        simulation_plot_config['dipole_scaling'] = 1.
```

Comment:
Is this line needed anymore? The `data_scaling.value` you are now using should be 1 already, correct?

Reply:
You are correct. I removed the line